### PR TITLE
[Network] Adopt XAMCORE_4_0 changes in .NET.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: src/Network/NWProtocolDefinition.cs [hunk 1]
```diff
@@ -103,7 +103,7 @@ public bool Equals (object other)
 		[DllImport (Constants.NetworkLibrary)]
 		static extern OS_nw_protocol_definition nw_protocol_copy_ws_definition ();
 
-#if !XAMCORE_4_0
+#if !NET
 #if !NET
```
Comment (tj-devel709) [inline src/Network/NWProtocolDefinition.cs:107]: Double `#if !NET` looks a little fishy to me

**Review by chamons — APPROVED** [review]
Approved, with a "look at TJ's question first".

File: src/Network/NWProtocolMetadata.cs [hunk 2]
```diff
@@ -50,8 +50,8 @@ public static NWProtocolMetadata CreateIPMetadata ()
 		[DllImport (Constants.NetworkLibrary)]
 		internal static extern OS_nw_protocol_metadata nw_udp_create_metadata ();
 
-#if !XAMCORE_4_0
-		[Obsolete ("Use the 'NSUdpMetadata' class and methods instead.")]
```
Comment (mandel-macaque) [inline src/Network/NWProtocolMetadata.cs:54]: lol

**Review by mandel-macaque — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated (https://github.com/xamarin/maccore/issues/581).

**Pull request merged**